src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateClone, action.extraData);
+      result.push({ ...stateClone });
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete stateClone[key];
+      }
+      result.push({ ...stateClone });
+    }
+
+    if (action.type === 'clear') {
+      for (const key in stateClone) {
+        delete stateClone[key];
+      }
+      result.push({ ...stateClone });
```

Comment:
Currently, the 'stateClone' is being mutated across different actions which can lead to unexpected behavior. Instead, you should create a new copy of the 'stateClone' inside each case of the action handling code to ensure that each action operates on a separate state clone.